src/cmd/services/m3dbnode/config/bootstrap.go

```diff
@@ -43,16 +43,23 @@ import (
 var (
 	// defaultNumProcessorsPerCPU is the default number of processors per CPU.
 	defaultNumProcessorsPerCPU = 0.125
+
+	// order in which bootstrappers are run
+	// (run in ascending order of precedence)
+	orderedBootstrappers = []string{
+		// Filesystem bootstrapping must be first.
+		bfs.FileSystemBootstrapperName,
+		// Peers and commitlog must come before the uninitialized topology bootrapping.
+		peers.PeersBootstrapperName,
+		commitlog.CommitLogBootstrapperName,
+		uninitialized.UninitializedTopologyBootstrapperName,
+	}
 )
 
 // BootstrapConfiguration specifies the config for bootstrappers.
 type BootstrapConfiguration struct {
```

Comment:
So one last thing here, can we add a config option to allow for peers to come in front of commit log? Uber I know need to continue using the peers bootstrapper in front of commit log bootstrapper.

What we can do is add an "enum" config option that allows for a much smaller set of configurability than the full list allowed.

```go
type BootstrapMode uint

const (
  DefaultBootstrapMode BootstrapStrategy = iota
  PreferPeersBootstrapMode
)

type BootstrapConfiguration struct {
  BootstrapMode BootstrapMode `yaml:"mode"`

 //...
}
```

And then we need to add the respective UnmarshalYAML functions, etc:
https://github.com/m3db/m3/blob/81c3e195f992c93f542dc44388bef051ba6591f1/src/dbnode/topology/consistency_level.go#L29-L114

I would make the representative string values things like "default" and "prefer_peers". We will need to allow someone not specifying anything for it to default to "default" which we can do specifically in the UnmarshalYAML function (by if value is empty string, we default to returning DefaultBootstrapMode).

Reply:
Sounds good. Done